packages/artillery-plugin-publish-metrics/lib/open-telemetry/tracing/base.js

```diff
@@ -144,10 +148,21 @@ class OTelTraceBase {
 
   async cleanup() {
     while (this.pendingRequestSpans > 0 || this.pendingScenarioSpans > 0) {
```

Comment:
Could we put a timeout on cleanup here? In case one of the conditions never evaluates to `true`. (This should really be done properly done at the runner level, but for now here is ok.)